src/main/java/com/powsybl/openloadflow/dc/DcTargetVector.java

```diff
@@ -24,7 +25,12 @@ public class DcTargetVector extends TargetVector<DcVariableType, DcEquationType>
     public static void init(Equation<DcVariableType, DcEquationType> equation, LfNetwork network, double[] targets) {
         switch (equation.getType()) {
             case BUS_TARGET_P:
-                targets[equation.getColumn()] = network.getBus(equation.getElementNum()).getTargetP();
+                LfBus bus = network.getBus(equation.getElementNum());
+                targets[equation.getColumn()] = bus.getTargetP();
+                // Only used for multi slack (BUS_TARGET_P equation is disabled for first slack bus)
+                if (bus.isSlack()) {
+                    targets[equation.getColumn()] += DcLoadFlowEngine.getActivePowerMismatch(network.getBuses()) / network.getSlackBuses().size();
```

Comment:
compute this value only once at the beginning of the init method e.g.

```java
final double slackActivePower = DcLoadFlowEngine.getActivePowerMismatch(network.getBuses()) / network.getSlackBuses().size();
```

Reply:
nononon I am wrong, forget it

Reply:
In my comprehension, this method is called for each equation. Which means that this value would be calculated for each equation instead of just for the multi-slack ones as it is in the current version

Reply:
oh right sorry